crates/aptos/src/node/mod.rs

```diff
@@ -0,0 +1,250 @@
+            SetOperator(tool) => tool.execute_serialized().await,
+        }
+    }
+}
+
+/// Stake coins for an account to the stake pool
+#[derive(Parser)]
+pub struct AddStake {
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+    /// Amount of coins to add to stake
+    #[clap(long)]
+    pub amount: u64,
+}
+
+#[async_trait]
+impl CliCommand<Transaction> for AddStake {
+    fn command_name(&self) -> &'static str {
+        "AddStake"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<Transaction> {
+        self.txn_options
+            .submit_script_function(
+                AccountAddress::ONE,
+                "Stake",
+                "add_stake",
+                vec![],
+                vec![bcs::to_bytes(&self.amount)?],
+            )
+            .await
+    }
+}
+
+/// Unlock staked coins
+///
+/// Coins can only be unlocked if they no longer have an applied lockup period
+#[derive(Parser)]
+pub struct UnlockStake {
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+    /// Amount of coins to unlock
+    #[clap(long)]
+    pub amount: u64,
+}
+
+#[async_trait]
+impl CliCommand<Transaction> for UnlockStake {
+    fn command_name(&self) -> &'static str {
+        "UnlockStake"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<Transaction> {
+        self.txn_options
+            .submit_script_function(
+                AccountAddress::ONE,
+                "Stake",
+                "unlock",
+                vec![],
+                vec![bcs::to_bytes(&self.amount)?],
+            )
+            .await
+    }
+}
+
+/// Withdraw all unlocked staked coins
+///
+/// Before calling `WithdrawStake`, `UnlockStake` must be called first.
+#[derive(Parser)]
+pub struct WithdrawStake {
+    #[clap(flatten)]
+    pub(crate) node_op_options: TransactionOptions,
+}
+
+#[async_trait]
+impl CliCommand<Transaction> for WithdrawStake {
+    fn command_name(&self) -> &'static str {
+        "WithdrawStake"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<Transaction> {
+        self.node_op_options
+            .submit_script_function(AccountAddress::ONE, "Stake", "withdraw", vec![], vec![])
+            .await
+    }
+}
+
+/// Increase lockup of all staked coins in an account
+#[derive(Parser)]
+pub struct IncreaseLockup {
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+    /// Number of seconds to increase the lockup period by
+    #[clap(long)]
+    pub(crate) new_lockup_secs: u64,
+}
+
+#[async_trait]
+impl CliCommand<Transaction> for IncreaseLockup {
+    fn command_name(&self) -> &'static str {
+        "IncreaseLockup"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<Transaction> {
+        self.txn_options
+            .submit_script_function(
+                AccountAddress::ONE,
+                "Stake",
+                "increase_lockup",
+                vec![],
+                vec![bcs::to_bytes(&self.new_lockup_secs)?],
```

Comment:
Should we do some simple validation here such as new_lockup_secs > current timestamp in secs?